### PR TITLE
[W5.7][W14-2] Lai Keen Lok

### PR DESCRIPTION
**Pull request by keenlok**

### CODE REVIEW

**Comment by YuPeiHenry** [comment]
Is this for W5.7?

**Comment by YuPeiHenry** [comment]
Good job at keeping incremental commits! For the updating of JUnit tests, on line 128 of AddCommandTest.java, assertTrue passes now because the command returns a list of people. However, the test did not verify that this list is indeed what is expected. Maybe improve the test by verifying the list of people from result.getRelevantPersons()?

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 1]
```diff
@@ -43,4 +54,12 @@ private void assertAreUnique(Object... objects) {
...
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+    
+    private void assertNotNull(Object... objects) {
+        assertFalse(Utils.isAnyNull(objects));
+    }
+    
+    private void assertHasNull(Object... objects) {
+        assertTrue(Utils.isAnyNull(objects));
+    }
```
Comment (kexiaowen) [inline test/java/seedu/addressbook/common/UtilsTest.java:64]: This pull request should only contain LO 5.7 stuff. The changes in this file are from last week's LO and should not be included.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -111,10 +112,14 @@ private CommandResult executeCommand(Command command)  {
...
-            ui.showToUser(e.getMessage());
-            throw new RuntimeException(e);
+        } catch (StorageOperationException soe) {
+            ui.showToUser(soe.getMessage());
+            return new CommandResult(String.format(Messages.MESSAGE_FILE_READ_ONLY, storage.getPath()));
         }
+//        catch (Exception e) {
+//            ui.showToUser(e.getMessage());
+//            throw new RuntimeException(e);
+//        }
```
Comment (kexiaowen) [inline src/seedu/addressbook/Main.java:122]: Do not keep code that has been commented out

File: src/seedu/addressbook/common/Messages.java [hunk 3]
```diff
@@ -15,4 +15,6 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_FILE_READ_ONLY = "Error! File cannot be overwritten. Please change the \n" +
+            "permissions of the file %1$s and try again!";
```
Comment (kexiaowen) [inline src/seedu/addressbook/common/Messages.java:19]: When wrapping lines, break before an operator. The + sign should be on line 19 instead.

**Review by kexiaowen — APPROVED** [review]
@keenlok Good attempt Keen Lok. You may add a description for your future PRs so that it is easier for reviewers to understand what your PR does quickly. You can close this PR after reading my comments.

**Pull request closed without merging**